packages/fxa-auth-server/test/local/routes/validators.js

```diff
@@ -386,17 +411,48 @@ describe('lib/routes/validators:', () => {
     });
 
     it('accepts unexpected keys', () => {
-      const res = subject.validate({
-        'capabilities:8675309': '123done,321done',
-        newThing: 'this is unexpected',
-      });
+      const res = subject.validate(
+        Object.assign({}, validProductMetadata, {
+          'capabilities:8675309': '123done,321done',
+          newThing: 'this is unexpected',
+        })
+      );
       assert.ok(!res.error);
     });
 
     it('rejects expected keys with invalid values', () => {
-      const res = subject.validate({
-        iconURL: true,
-      });
+      const res = subject.validate(
+        Object.assign({}, validProductMetadata, {
+          webIconURL: true,
+        })
+      );
+      assert.ok(res.error);
+    });
+
+    it('rejects if missing required props', () => {
+      let res = deletePropAndValidate('downloadURL');
+      assert.ok(res.error);
+
+      res = deletePropAndValidate('product:privacyNoticeURL');
+      assert.ok(res.error);
+
+      res = deletePropAndValidate('product:privacyNoticeDownloadURL');
+      assert.ok(res.error);
+
+      res = deletePropAndValidate('product:termsOfServiceURL');
+      assert.ok(res.error);
+
+      res = deletePropAndValidate('product:termsOfServiceDownloadURL');
+      assert.ok(res.error);
+    });
+
+    it('rejects if no capabilities set', () => {
+      const res = subject.validate(
+        Object.assign({}, validProductMetadata, {
+          capabilities: null,
+          'capabilities:aFakeClientId12345': null,
+        })
```

Comment:
This looks like a 'rejects expected keys with invalid values' test since null is an invalid value for capabilities.

Reply:
removed, this was old test, should be tested above with the deletePropAndValidate method